control-plane/csi-driver/src/bin/node/nodeplugin_svc.rs

```diff
@@ -177,3 +104,67 @@ pub(crate) async fn find_mount(
         Ok(None)
     }
 }
+
+/// /// Issue FIFREEZE and FITHAW ioctl on the mount_path derived from the volume_id based on the
+/// command.
+pub(crate) async fn fsfreeze(volume_id: &str, command: FsFreezeOpt) -> Result<(), ServiceError> {
+    let uuid = Uuid::parse_str(volume_id).context(InvalidVolumeId {
+        volume_id: volume_id.to_string(),
+    })?;
+
+    if let Some(device) = Device::lookup(&uuid).await.context(InternalFailure {
+        volume_id: volume_id.to_string(),
+    })? {
+        let device_path = device.devname();
+        if let Some(mnt) = mount::find_mount(Some(&device_path), None) {
+            // Make a preflight check, to ensure subsystem has atleast one live path.
+            let device_nqn = device.devnqn();
+            fsfreeze_preflight_check(volume_id, &device_nqn).map_err(|errno| {
+                ServiceError::FsfreezeFailed {
+                    volume_id: volume_id.to_string(),
+                    errno,
+                }
+            })?;
+
+            // Execute the fsfreeze call.
+            let dest = mnt.dest.display().to_string();
+            return fsfreeze_ioctl(command, dest).await.map_err(|errno| {
```

Comment:
fsfreeze_ioctl should take a &Path I guess?

Reply:
Have tried that, but seems it causing lifetime issue.